File: Question_3/election_system.py
def process_vote(voters, candidates, vote):
    voter = next((v for v in voters if v["citizen_id"] == vote["citizen_id"]), None)
    if not voter:
        return "Fraud detected"
    if voter["vote_status"]:
        return "Citizen has already voted"
    candidate = next((c for c in candidates if c["candidate_id"] == vote["candidate_id"]), None)
    if not candidate:
        return "Invalid candidate ID"
    candidate["count"] += 1
    voter["vote_status"] = True
    return candidates
voters = [
    {"citizen_id": 1, "vote_status": False},
    {"citizen_id": 2, "vote_status": False}
]
candidates = [
    {"candidate_id": "A", "count": 0},
    {"candidate_id": "B", "count": 0}
]
vote = {"citizen_id": 1, "candidate_id": "A"}
result = process_vote(voters, candidates, vote)
print(result)

File: Question_3/electionSystem.kt
data class Voter(var citizenId: String, var voteStatus: Boolean)
data class Candidate(val candidateId: String, var count: Int)
data class Vote(val citizenId: String, val candidateId: String)
fun processVote(
    voters: MutableList<Voter>,
    candidates: MutableList<Candidate>,
    vote: Vote
): Any {
    val voter = voters.find { it.citizenId == vote.citizenId }
    if (voter == null) {
        return "Fraud detected"
    }
    if (voter.voteStatus) {
        return "Citizen has already voted"
    }
    val candidate = candidates.find { it.candidateId == vote.candidateId }
    if (candidate == null) {
        return "Invalid candidate ID"
    }
    candidate.count += 1
    voter.voteStatus = true
    return candidates
}
fun main() {
    val voters = mutableListOf(
        Voter("123", false),
        Voter("456", false)
    )
    val candidates = mutableListOf(
        Candidate("A", 0),
        Candidate("B", 0)
    )
    val vote = Vote("123", "A")
    val result = processVote(voters, candidates, vote)
    println(result)
}






